SharedControls/Viewers/BuildingViewer.xaml.cs

```diff
@@ -15,41 +15,63 @@ public partial class BuildingViewer : Window, IAssignmentViewer
         public event Action<IAssignment, bool> CompletionStateChanged;
 
         private readonly BuildingAssignment _assignment;
+        private int _currentItemIndex = 0;
 
         public string TaskTitle { get; }
+
         public BuildingViewer(BuildingAssignment assignment)
         {
             InitializeComponent();
             DataContext = this;
 
             _assignment = assignment;
-
             TaskTitle = _assignment.Title;
 
-            foreach (var item in _assignment.Items)
+            // Загрузка первого выражения
+            LoadCurrentItem();
+        }
+
+        // Метод для загрузки текущего элемента
+        private void LoadCurrentItem()
+        {
+            spItems.Children.Clear();
+
+            if (_currentItemIndex < _assignment.Items.Count)
             {
+                var item = _assignment.Items[_currentItemIndex];
                 var buildingItemViewControl = new BuildingItemViewControl(item) { Tag = item.Text };
                 spItems.Children.Add(buildingItemViewControl);
             }
+            _currentItemIndex++;
         }
+
         private void btnCheck_Click(object sender, RoutedEventArgs e)
         {
-            foreach (BuildingItemViewControl buildingItemViewControl in spItems.Children)
+            if (spItems.Children.Count == 0) return;
+
+            var buildingItemViewControl = (BuildingItemViewControl)spItems.Children[0];
+            var arrangedPhrase = GetUserArrangedPhrase(buildingItemViewControl);
+
+            // Проверка правильности
+            if (arrangedPhrase != buildingItemViewControl.Tag.ToString())
             {
-                var arrangedPhrase = GetUserArrangedPhrase(buildingItemViewControl);
+                CompletionStateChanged?.Invoke(_assignment, false);
+                MessageBox.Show(Translations.GetValue("Incorrect"));
+                return;
+            }
 
-                // Check if the user input matches one of the options
-                if (arrangedPhrase != buildingItemViewControl.Tag.ToString())
-                {
-                    CompletionStateChanged?.Invoke(_assignment, false);
-                    MessageBox.Show(Translations.GetValue("Incorrect"));
-                    return;
-                }
+            // Проверка на завершение всех элементов
+            if (_currentItemIndex >= _assignment.Items.Count)
```

Comment:
you only need to check for equality, when currentItemIndex equals Items.Count it means you have already reached the end, for instance, if you have two phrases, Items.Count will be 2, but indexes start from 0, so when currentItemIndex is 2 it means you already loaded both phrases and can close with success